src/main/java/org/folio/rest/support/StatusUpdatedDateGenerator.java

```diff
@@ -1,11 +1,11 @@
 package org.folio.rest.support;
 
-import java.text.SimpleDateFormat;
-import java.util.Date;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
 
 public final class StatusUpdatedDateGenerator {
+    private final static DateTimeFormatter df = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
     public static String generateStatusUpdatedDate() {
-      SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
-      return sdf.format(new Date());
-    } 
+      return df.format(ZonedDateTime.now());
+    }
```

Comment:
Why has this been changed?

Is there a significant computation saving from the static field (I would think the compiler would have already optimised this away)?

Reply:
During my analysis of memory allocations during my test runs, large percentage of allocations were from  SimpleDateFormat instantiated by this method. This change is meant to reduce these allocations by using a single object that format dates. This reduces GC clean up over time(during multi hour data import jobs), I did not notice substantial change in response times of a single request.

Reply:
Most computation in code written by us is used by regex pattern matching for publication dates and spawning the new dates for status update field on the instance.

Reply:
> During my analysis of memory allocations during my test runs, large percentage of allocations were from SimpleDateFormat instantiated by this method.

That is useful to know, thank you.

> This change is meant to reduce these allocations by using a single object that format dates. This reduces GC clean up over time(during multi hour data import jobs), I did not notice substantial change in response times of a single request.

Thanks, that makes sense.